### PR TITLE
Upgrade all dependencies and shrink compiled package

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,33 +26,29 @@
   "author": "GoogleCloudPlatform",
   "license": "Apache-2.0",
   "dependencies": {
-    "@actions/core": "^1.5.0",
-    "fs": "0.0.1-security",
-    "google-auth-library": "^7.9.2",
-    "uuidv4": "^6.2.12"
+    "@actions/core": "^1.6.0",
+    "google-auth-library": "^7.10.3",
+    "yaml": "^1.10.2"
```
Comment (sethvargo) [inline package.json:31]: YAML was a dev dep, but it's actually used in the gkeClient

File: src/gkeClient.ts [hunk 2]
```diff
@@ -45,7 +49,7 @@ type ClientOptions = {
  */
 export class ClusterClient {
   readonly defaultEndpoint = 'https://container.googleapis.com/v1';
-  readonly userAgent = 'github-actions-get-gke-credentials/0.1.0';
+  readonly userAgent = `github-actions-get-gke-credentials/${appVersion}`;
```
Comment (sethvargo) [inline src/gkeClient.ts:52]: Minor enhancement here

File: src/main.ts [hunk 3]
```diff
@@ -43,20 +49,16 @@ async function run(): Promise<void> {
...
 
     // Export KUBECONFIG env var with path to kubeconfig
-    core.exportVariable('KUBECONFIG', kubeConfigPath);
-    core.info('Successfully created and exported "KUBECONFIG"');
-  } catch (error) {
-    core.setFailed(error.message);
+    exportVariable('KUBECONFIG', kubeConfigPath);
+    logInfo('Successfully created and exported "KUBECONFIG"');
+  } catch (err) {
+    setFailed(`google-github-actions/get-gke-credentials failed with: ${err}`);
```
Comment (sethvargo) [inline src/main.ts:61]: Updated this error message to clearly prefix the action

File: src/util.ts [hunk 4]
```diff
@@ -14,9 +14,9 @@
  * limitations under the License.
  */
 
-import * as fs from 'fs';
-import path from 'path';
-import { v4 as uuidv4 } from 'uuid';
+import { promises as fs } from 'fs';
+import { join as pathjoin } from 'path';
+import { randomBytes } from 'crypto';
```
Comment (sethvargo) [inline src/util.ts:19]: Dropped uuid dep

File: src/util.ts [hunk 5]
```diff
@@ -29,11 +29,18 @@ export async function writeFile(fileContent: string): Promise<string> {
...
+  const uniqueName = randomBytes(12).toString('hex');
+  const kubeConfigPath = pathjoin(workspace, uniqueName);
+
   try {
-    await fs.writeFileSync(kubeConfigPath, fileContent);
+    await fs.writeFile(kubeConfigPath, fileContent);
     return kubeConfigPath;
   } catch (err) {
-    throw new Error(`Unable to write kubeconfig to file: ${kubeConfigPath}`);
+    throw new Error(`Failed to write kubeconfig to ${kubeConfigPath}: ${err}`);
```
Comment (sethvargo) [inline src/util.ts:44]: Twas bug: we were swallowing the root cause of the error

File: tests/util.test.ts [hunk 6]
```diff
@@ -14,54 +14,42 @@
  * limitations under the License.
  */
 
-import * as chai from 'chai';
-import chaiAsPromised from 'chai-as-promised';
+import { expect } from 'chai';
 import 'mocha';
-import * as sinon from 'sinon';
```
Comment (sethvargo) [inline tests/util.test.ts:20]: I dropped sinon and interact directly with the filesystem. It's a better test and actually somehow faster in my benchmarks. Everything happens in tmpdir, so it's cleaned automatically by the OS too

File: package.json (hunk 1 above)
Comment (bharathkkb) [reply package.json:31]: huh, weird it didnt complain

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**